AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -605,7 +635,7 @@ - (ASSizeRange)dataController:(ASDataController *)dataController constrainedSize
   }
   
   if (_asyncDelegateImplementsInsetSection) {
-    sectionInset = [_asyncDelegate collectionView:self layout:self.collectionViewLayout insetForSectionAtIndex:indexPath.section];
+    sectionInset = [(id<ASCollectionViewDelegateFlowLayout>)_asyncDelegate collectionView:self layout:self.collectionViewLayout insetForSectionAtIndex:indexPath.section];
```

Comment:
I mostly understand, but not sure if completely understand.  This is still calling out to the _asyncDelegate; so are you saying that you plan to keep this as a delegate method, just generally deprecate / discourage its use and attempt to make its specificity to flow layouts more clear by creating a new sub-protocol?  That would make sense to me, if the protocol in question derives from ASCollectionViewDelegate.

The name doesn't really feel right because it doesn't end in Delegate, although I do see the logic behind it.  We should consider ASCollectionViewFlowLayoutDelegate but I trust your choice here.